src/readers/morphologyHDF5.cpp

```diff
@@ -70,193 +87,154 @@ Property::Properties load(const HighFive::Group& group) {
-        const auto majorVersion = _properties._cellLevel.majorVersion();
-        const auto minorVersion = _properties._cellLevel.minorVersion();
-        if (majorVersion == 1 && (minorVersion == 1 || minorVersion == 2)) {
-            uint32_t family;
-            const auto familyAttr = metadata.getAttribute(_a_family);
-            familyAttr.read(family);
-            _properties._cellLevel._cellFamily = static_cast<CellFamily>(family);
-        } else {
-            throw morphio::RawDataError(
-                "Error in " + source + "\nUnsupported h5 version: " + std::to_string(majorVersion) +
-                "." + std::to_string(minorVersion) +
-                "See "
-                "https://bbpteam.epfl.ch/documentation/projects/Morphology%20Documentation/latest/"
-                "index.html for the list of supported versions.");
-        }
-    } catch (const HighFive::Exception&) {
-    }
-
-    try {
-        _resolveV1();
-
-        // The fact that the resolution passes and majorVersion is still uninitialized tells us
-        // it is h5v1.0
-        if (_properties._cellLevel.majorVersion() == 0) {
-            std::get<1>(_properties._cellLevel._version) = 1;  // major version
-            std::get<2>(_properties._cellLevel._version) = 0;  // minor version
-            // Version 1.0 only support NEURON has a CellFamily.
-            // Other CellFamily have been added in version 1.1:
-            // https://bbpteam.epfl.ch/documentation/projects/Morphology%20Documentation/latest/h5v1.html
-            _properties._cellLevel._cellFamily = CellFamily::NEURON;
+    // default to h5v1.0
+    uint32_t majorVersion = 1;
+    uint32_t minorVersion = 0;
+    _properties._cellLevel._cellFamily = CellFamily::NEURON;
+
+    if (!_group.exist(_d_points) || !_group.exist(_d_structure)) {
+        // h5v2 is deprecated, but it can be detected, throw a custom error messages if it is
+        if (_group.exist(_g_v2root)) {
+            throw RawDataError(
+                "Error in " + source +
+                "\nh5v2 is no longer supported, see: https://github.com/BlueBrain/MorphIO#H5v2");
         }
-        return;
-    } catch (const HighFive::Exception&) {
+        throw RawDataError("Missing " + _d_points + " or " + _d_structure +
+                           " datasets, cannot load morphology without them");
     }
 
-
-    // h5v2 attempt
-    {
-        uint32_t potential_h5v2_version;
-        try {
-            _group.getGroup(_g_root).getAttribute(_a_version).read(potential_h5v2_version);
-        } catch (const HighFive::Exception&) {
+    // if there is metadata, perhaps it's h5v1, minor version 1, 2
+    if (_group.exist(_g_metadata)) {
+        const auto metadata = _group.getGroup(_g_metadata);
+        if (metadata.hasAttribute(_a_version)) {
+            const auto attr = metadata.getAttribute(_a_version);
+
+            std::array<uint32_t, 2> versions;
```

Comment:
This is not initialized and I am not familiar with HighFive but shouldn't we init to something like : 0, 0 ?